### PR TITLE
SimpleModal can be under or on top of Navigation

### PR DESCRIPTION
**Pull request by gisete**
Adding a new prop to the SimpleModal to control if the modal can be on top of the navigation or not.

## Description

The new prop `alwaysOnTop` can be set to`true` if we want the SimpleModal to always be on top of all other elements including the navigation.

## Motivation and Context

The Plan Selector takes a lot of space and for smaller screens the navigation was hiding the close button and the top part of the modal. We want to have the option of making the modal the only focal point for the customer.

## Screenshots (if appropriate):
With prop:
image

Without prop:
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style and guidelines of this project. 
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated the **CHANGELOG** document.
- [x] I have added tests to cover my changes.
- [x] I have performed a self-review of my own code
- [x] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] All new and existing tests passed.
- [x] I have performed the accessibility audit of my UI changes according to the accessibility doc. 
- [ ] [Buffer Engineers] Someone from the Design team reviewed and approved my changes
- [ ] [Buffer Engineers] I have notified the BDS team of my changes in the #proj-design-system Slack channel

### CODE REVIEW

**Review by federicoweber — CHANGES_REQUESTED** [review]
Hey @gisete, thanks for working on this.
I'm a bit hesitant to merge this in for two reasons:
1. it's coupling the modals style, to the business logic
2. it's not fully solving the issue, as it still allows the modals to block the navigation.

---

I was playing with a few options quickly and I think we can do something like this 
up to 810px of window height, we can set the `Container`  `min-height: 810px;`, that will allow us to properly render the payment modals up to a screen size of 1280px. 

image

For anything less than that, we can set `Container` `overflow: auto;` and set it's first children `min-height` in such a way that we can fully scroll the content ( in this case the `AnimationWrapper`, but we don't want to set the rule on the `AnimationWrapper` directly because it's used in other cases as well 😉 ).

min height modal 

`Container` `min-height: 810px; overflow: auto` can be the default values, but we will probably need to limit the `AnimationWrapper` `min-heigh` with a media query to avoid layout artifacts 🙂 .

How does that look to you?
Please let me know if you have any question or if I can help with anything 😊

Thanks!

**Comment by gisete** [comment]
Hey @federicoweber, sorry, I realize I tagged you in the original Slack thread but that one is missing the whole context. What we discussed later is that we do want the Plan Selector to be on top of the navigation. [More about it here.](https://buffer.slack.com/archives/C0DLE125Q/p1648488657225729) 

I know you made these changes because of the paywall but do you think we could make this work for the Plan Selector only?

**Comment by federicoweber** [comment]
Thanks for the extra context @gisete, that make sense one.

One thing I will still suggest to add in the `min-height` for the `Container`, as that will improve the positioning of the modal, minimizing the cases where that's overlapping with the navbar.

Without min-height:
image

With min-height:
image

**Review by federicoweber — APPROVED** [review]
Thanks @gisete 🙌 🌮 :shipit:

File: src/components/SimpleModal/style.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ export const Container = styled.div`
   position: fixed;
   box-sizing: border-box;
   width: 100%;
-  z-index: 9999;
+  z-index: ${({NonDismissibleModal}) => NonDismissibleModal? '1' : '9999' };
```
[the remark this answers is not in the document]
Comment (gisete) [reply src/components/SimpleModal/style.js:24]: Good question, I just restored to what it was there before 🙂  It does seem hight but I wasn't sure what value other elements are using. One thing I can think of is that Pendo elements are quite high too so we want to always be on top of that.

**Pull request merged**